### ISSUE TITLE
Field 'id' expected a number but got '41jl7ngByXlYCcE8rXBp' with to_queryset()

### ISSUE THREAD

**Issue by Sachin-Kahandal** [django-es/django-elasticsearch-dsl#341]
Somethings up with id, though I haven't made any changes in to_queryset function

s = MyntraClothesDocument.search(index="myntra_clothes").query(MultiMatch(query=text, fields=['Title', 'Description'], operator="AND", fuzziness='AUTO'))
qs = s.to_queryset()

ValueError: Field 'id' expected a number but got '41jl7ngByXlYCcE8rXBp'.

Thank you

**Comment by KunstDerFuge** [comment]
Same thing just happened to me -- I believe it was triggered by doing an incremental update... or at least it happened after doing one:
```python
    posts_created = Post.objects.bulk_create(new_posts, ignore_conflicts=True)
    PostDocument().update(posts_created)
```

All IDs in the index now appear to be these weird hashes rather than ints, so can't be used to get a QuerySet.

**Comment by safwanrahman** [comment]
Interesting. Need to look at it. @saadmk11 Will you have some time to look?

**Comment by saadmk11** [comment]
@KunstDerFuge Thanks for reporting the issue. Directly using the objects list returned by the `bulk_create()` can cause this issue.

From Django's (`bulk_create()`) Documentation:

> If the model’s primary key is an AutoField, the primary key attribute can only be retrieved on certain databases (currently PostgreSQL and MariaDB 10.5+). On other databases, it will not be set.

Ref: https://docs.djangoproject.com/en/3.2/ref/models/querysets/#bulk-create

Please let us know which Database and Django version you are uisng.

**Comment by KunstDerFuge** [comment]
Yes, I had heard this, and that's what's confusing to me. I'm using Postgres 12.8 and Django 3.2.3.

**Comment by KunstDerFuge** [comment]
I just ran a simple test, bulk creating three objects and checking what is returned from `bulk_create()`. It is a list of objects where both `object.id` and `object.pk` contain correct, auto-incremented integer keys. So perhaps in my case it wasn't due to `bulk_create()`. That's quite confusing...

**Comment by saadmk11** [comment]
After a little bit of research, I have found that if we use `ignore_conflicts=True` with `bulk_create()` it does not set the `pk` to the returned objects. That's why this is happening. If you don't use `ignore_conflicts=True` it should work as expected.

Django Ticket on this issue: https://code.djangoproject.com/ticket/30138
